File: cpp/746_使用最小花费爬楼梯_easy.cpp
class Solution {
public:
    int minCostClimbingStairs(vector<int>& cost) {
        int left = cost[0];
        int right = cost[1];
        for(int i=2; i<cost.size(); i++){
            int tmp = min(left, right);
            left = right;
            right = tmp + cost[i];
        }
        return min(left, right);
    }
};

File: python/746_使用最小花费爬楼梯_easy.py
class Solution:
    def minCostClimbingStairs(self, cost: List[int]) -> int:
        left, right = cost[0], cost[1]
        for i in range(2, len(cost)):
            right, left = min(left + cost[i], right + cost[i]), right
        return min(left, right)